src/katsdpcalproc/pointing.py

```diff
@@ -1,87 +1,87 @@
-    offset_end_times : sequence of *N* floats
-        Unix timestamp at the end of each pointing track
-    track_duration : float
-        Duration of each pointing track, in seconds
+    bp_gains : a list containing a numpy array of shape (no.channels,
+        no.polarizations,no.antennas) for each pointing offset *N*, ie.
+        for N pointing offsets, len(bp_gains)=N.
+    gains : a list containing a numpy array of shape
+        (no.polarizations,no.antennas) for each pointing offset *N*,
+        ie. for N pointing offsets, len(gains)=N.
+    offsets : list containing the requested (x, y) pointing offsets
+        relative to target, in degrees.
+    track_duration : float, Duration of each pointing track, in seconds
+    NUM_CHUNKS: Group the frequency channels into this many sections to
+        obtain pointing fits (default =16)
+    ants: A list containing <class: katpoint.Antenna> objects for each
+        antenna used in the observation
+    centre_freq, bandwidth, no_channels: floats, centre frequency, bandwidth
+        and number of frequency channels
+    pols: list, A list containing polarisations, eg, ["h","v"]
 
     Returns
     -------
     data_points : dict mapping receptor index to (x, y, freq, gain, weight) seq
-        Complex gains per receptor, as multiple records per offset and frequency
-
+            Complex gains per receptor, as multiple records per offset and
+            frequency chunk ie. len(data_points)=63, len(data_points[i])=
+            Num_chunks*no.offsets, len(data_points[i][j]=5)
     """
-    cal_channel_freqs = session.get_cal_channel_freqs()
-    chunk_freqs = cal_channel_freqs.reshape(NUM_CHUNKS, -1).mean(axis=1)
-    pols = session.telstate['cal_pol_ordering']
+
+    if no_channels % NUM_CHUNKS != 0:
+        raise NotMUltipleError("NUM_CHUNKS is not a multiple of number of channels")
+    # Calculating chunk frequencies
+
+    channel_freqs = centre_freq + ((np.arange(no_channels) - no_channels / 2)
+                                   * (bandwidth / no_channels))
+
+    chunk_freqs = channel_freqs.reshape(NUM_CHUNKS, -1).mean(axis=1)
     data_points = {}
-    # Iterate over offset pointings
-    for offset, offset_end in zip(offsets, offset_end_times):
-        offset_start = offset_end - track_duration
-        # Obtain interferometric gains per pointing from the cal pipeline
-        try:
-            bp_gains = session.get_cal_solutions('B', start_time=offset_start,
-                                                 end_time=offset_end)
-            gains = session.get_cal_solutions('G', start_time=offset_start,
-                                              end_time=offset_end)
-        except CalSolutionsUnavailable as err:
-            user_logger.warning('Skipping offset %s: %s', offset, err)
-            continue
-        # Iterate over receptors
-        for a, ant in enumerate(session.observers):
+
+    for i, e, f in zip(offsets, bp_gains, gains):
```

Comment:
One-letter names are confusing in general, except for indices and maths / physics formulas. `pylint` would also complain here.

`i` -> `offset`
`e` -> `offset_bp_gain`
`f` -> `offset_gain`